app/src/main/java/net/kwatts/powtools/RideDetailActivity.java

```diff
@@ -112,21 +122,38 @@ private void retrieveData(ArrayList<Entry> timeSpeedMap) {
                 if (moment.getGpsLatDouble() != null && moment.getGpsLongDouble() != null) {
                     timeLocationMap.put(time, new LatLng(moment.getGpsLatDouble(), moment.getGpsLongDouble()));
                 }
-                Attribute attribute = database.attributeDao().getFromMomentAndKey(moment.id, "speed");
-                if (attribute != null && attribute.getValue() != null) {
-                    String value = attribute.getValue();
-                    timeSpeedMap.add(new Entry(time, Float.valueOf(value)));
+//                Attribute attribute = database.attributeDao().getFromMomentAndKey(moment.id, "speed");
+                List<Attribute> attributes = database.attributeDao().getFromMoment(moment.id);
+                for (Attribute attribute : attributes) {
+                    if (attribute != null
```

Comment:
This `null` check needed? If so, then probably better just to make sure and never save `null`s to the db.